src/governance.h

```diff
@@ -147,11 +163,19 @@ class CGovernanceManager
     void UpdatedBlockTip(const CBlockIndex *pindex);
     int64_t GetLastDiffTime() {return nTimeLastDiff;}
     void UpdateLastDiffTime(int64_t nTimeIn) {nTimeLastDiff=nTimeIn;}
+
+    int GetCachedBlockHeight() { return nCachedBlockHeight; }
```

Comment:
Yes, but 1) `UpdateCachedBlockHeight()` is called in way too many places -> more `cs_main` locks 2) this job is already been done by `UpdatedBlockTip()` which is triggered by `dsnotificationinterface` via corresponding signal and it's triggered only when tip actually changes.